mip/solver.go

```diff
@@ -1,100 +1,13 @@
 package mip
 
-import "time"
-
-// Solver exposes an API to run a MIP solver
-//
-//			model := mip.NewModel()
-//
-//	 	// build the model
-//
-//			provider := "my_favorite_solver"
-//
-//			solver, err := NewSolver(provider, model)
-//
-//			solution, err := solver.Solve(mip.DefaultSolverOptions())
+// Solver for a MIP problem.
 type Solver interface {
 	// Solve is the entrypoint to solve the model associated with
 	// the invoking solver. Returns a solution when the invoking solver
 	// reaches a conclusion.
 	Solve(options SolveOptions) (Solution, error)
 }
 
-// SolveOptions interface to options for back-end solver.
-type SolveOptions interface {
```

Comment:
I remember that we back in the day aimed to support options that we expected to be available on all solvers via common functionality. It appears that we are dropping that. Are we aligned about this? Also, this is breaking, no? :thinking: 
I could be missing a bigger point though. :relaxed: 
cc @davidrijsman @dirkschumacher

Reply:
That is not dropped. The options that are "generally" available were the solver duration, verbosity, absolute gap and relative gap. Those 4 options were the only things you could control. The custom ones had an API but it wasn't connected to anything so you couldn't configure anything else. To preserve that "generally available" characteristic, I kept those 4 options as the ones you could set with the CLI for any provider. I honestly was too lazy to compare highs and xpress and figure out more things they have in common. Lastly, the custom params for each provider can now be set and they work because they are now connected. You just need to know the name, type and the value you want to set.